addons/account/models/account_payment.py

```diff
@@ -853,6 +860,15 @@ def _check_payment_method_line_id(self):
             elif pay.payment_method_line_id.journal_id and pay.payment_method_line_id.journal_id != pay.journal_id:
                 raise ValidationError(_("The selected payment method is not available for this payment, please select the payment method again."))
 
+    @api.onchange('payment_method_line_id')
+    def _onchange_journal(self):
+        """
+            This onchange function will change the journal when selecting a partner if the partner has some payment
+            method set to put the journal associated to the payment method line.
+        """
+        if self.payment_method_line_id.journal_id:
+            self.journal_id = self.payment_method_line_id.journal_id
```

Comment:
This is a bad idea. Journal and company are already in a cyclic dependency loop and you are adding an onchange on top of that.

Reply:
Didn't find another solution that using an onchange since journal_id is a related field that is precomputed, if i replace the related for a compute, for the api.depends on payment_method_line to work i have to put it in precompute aswell, not sure we want that ?

Reply:
Let's try an inverse on the partner_id :smile: